src/Payment/Canceller/PaymentCanceller.php

```diff
@@ -0,0 +1,44 @@
+<?php
+
+declare(strict_types=1);
+
+namespace CommerceWeavers\SyliusTpayPlugin\Payment\Canceller;
+
+use CommerceWeavers\SyliusTpayPlugin\Payment\Exception\PaymentCannotBeCancelledException;
+use SM\Factory\FactoryInterface;
+use Sylius\Abstraction\StateMachine\Exception\StateMachineExecutionException;
+use Sylius\Abstraction\StateMachine\StateMachineInterface;
+use Sylius\Component\Core\Model\PaymentInterface;
+use Sylius\Component\Payment\PaymentTransitions;
+
+final class PaymentCanceller implements PaymentCancellerInterface
+{
+    public function __construct(
+        private readonly StateMachineInterface|null $stateMachine,
```

Comment:
```suggestion
        private readonly ?StateMachineInterface $stateMachine,
```

just to follow the convention